### PR TITLE
chapter_6

### PR DESCRIPTION
**Pull request by DaudKhadijah01**
ready

### CODE REVIEW

**Comment by DaudKhadijah01** [comment]
ready

**Review by ghost — COMMENTED** [review]
fix your conflict please

**Comment by DaudKhadijah01** [comment]
how to fix the conflict?

**Comment by DaudKhadijah01** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /* universal */
 
 body {
+
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: remove empty line

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
 padding-top: 60px;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: indent
ex:
```suggestion
body {
  padding-top: 60px;
}
```

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -18,10 +19,30 @@ text-align: center;
 
 .center h1 {
 margin-bottom: 10px;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: indent
```suggestion
  margin-bottom: 10px;
```

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,2 +1,8 @@
 module ApplicationHelper
+
+  def full_title page_title = ""
+   base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb]: - indent = 2 spaces
```suggestion
    base_title = "Ruby on Rails Tutorial Sample App"
```

- apply I18n to translate this text `"Ruby on Rails Tutorial Sample App"`
ex:
```suggestion
   base_title = t "rails_app_title"
```

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<header class="navbar navbar-fixed-top navbar-
+inverse">
+
+  <div class="container">
+   <%= link_to "sample app", "#", id: "logo" %>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb:5]: apply I18n for text `"sample app"`, fix the same in this file

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -16,6 +16,9 @@
 
     <div class="container">
       <%= yield %>
+
+    <%= yield %>
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: remove this line (it is duplicate with line 18)

File: test/fixtures/users.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/users.yml]: all of test files in tutorial are not necessary, so you can delete this file

File: test/models/user_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/models/user_test.rb]: delete this file

File: app/views/static_pages/help.html.erb [hunk 9]
```diff
@@ -1,12 +1,13 @@
 <% provide :title, "Help" %>
```
Comment (huongnt-2545) [inline app/views/static_pages/help.html.erb:1]: apply I18n to translate for all text in this file

**Review by huongnt-2545 — COMMENTED** [review]
Apply Rails I18n for project
refs: https://guides.rubyonrails.org/i18n.html#internationalization-and-localization

**Comment by DaudKhadijah01** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 10]
```diff
@@ -18,10 +18,29 @@ text-align: center;
...
+padding-top: 60px;
+}
+
+section {
+overflow: auto;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: indent
fix the same error in this file
```suggestion
  overflow: auto;
```

ex:
```
#Good
textarea {
  resize: vertical;
}
```

File: app/assets/stylesheets/custom.scss [hunk 11]
```diff
@@ -36,4 +55,20 @@ font-weight: bold;
 #logo:hover {
 color: #fff;
 text-decoration: none;
+
+  #logo {
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]: indent
```suggestion
#logo {
```

File: app/controllers/application_controller.rb [hunk 12]
```diff
@@ -1,2 +1,15 @@
 class ApplicationController < ActionController::Base
+
+  protect_from_forgery with: :exception
+  before_action :set_locale
+
+  private
+
+  def set_locale
+  locale = params[:locale] || I18n.default_locale
```
Comment (huongnt-2545) [inline app/controllers/application_controller.rb]: indent = 2 spaces
```suggestion
    locale = params[:locale] || I18n.default_locale
```

File: app/helpers/application_helper.rb [hunk 13]
```diff
@@ -1,2 +1,8 @@
 module ApplicationHelper
+
+  def full_title page_title = ""
+   base_title = t(".rails_app_title")
```
Comment (huongnt-2545) [inline app/helpers/application_helper.rb:4]: indent = 2 spaces

File: test/controllers/static_pages_controller_test.rb [hunk 14]
```diff
@@ -0,0 +1,8 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/static_pages_controller_test.rb]: delete file test because it is no need

**Review by huongnt-2545 — COMMENTED** [review]
This pull request has 3 commits, you need to combine in 1

**Comment by DaudKhadijah01** [comment]
ready

**Comment by DaudKhadijah01** [comment]
i try to apply i18n but it does not appear on my sample app, and when i try to do chapter 7 on the new branch, some of files from chapter 3 are missing. how to resolve it?

**Review by ghost — COMMENTED** [review]
This pull request has 4 commits, you need to combine in 1

**Comment by DaudKhadijah01** [comment]
ready

**Comment by DaudKhadijah01** [comment]
ready

**Comment by DaudKhadijah01** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**